### ISSUE TITLE
[Bug] failure to connect and app crash and bugcheck

### ISSUE THREAD

**Issue by Protonus** [OnewheelCommunityEdition/OWCE_App#75]
**Describe the bug**
Firstly, the "About this app" in the Google Play Store, does not list the Pint. It just says "V1, Plus, and XR".  AFAIK, my Pint's firmware is supported? If so this should probably be corrected.  

I have yet to get OWCE to successfully connect, ever.   The stock / OEM, FM OW app  has no trouble connecting.  I have sometimes received the "OWCE needs help to connect to your board" message trying to connect.  I have followed those instructions exactly as presented many times. I tried doing a force close on the OW app as well as clearing it from recents instead, after connecting and waiting for ride profile to load.  Neither worked on many attempts.  It has yet to connect even once. :( 

On one occasion, when the app tried to close itself, it crashed, and requested I send the crash dump which I did.

On two other occasions, after trying to connect, I got a message of: "OWCE has closed because this app has a bug". I've attached a screenshot. I've never seen this message in Android before.

**To Reproduce**
Open OWCE.
Click OW - only board in list, is my correct serial number. 
It either fails to connect or do anything (connection window disappears after a few seconds), or it crashes, or asks me to help it connect.  
If it asks for help connecting, I follow the instructions. But it fails to connect. Or crashes. 

**Expected behavior**
The OWCE app to connect to my Pint, and not crash / error. 

**Screenshots**
![Screenshot_20220521-223041_Device care](https://user-images.githubusercontent.com/502342/169675842-51261f8c-750a-4508-bec1-0e378f1c44a1.jpg)
![Screenshot_20220521-222726](https://user-images.githubusercontent.com/502342/169675844-587e7eef-2257-471d-bd76-e84f3d0b2920.jpg)

**Phone (please complete the following information):**
 - Device: Samsung Galaxy XCover Pro
 - OS: Android 
 - Version 12

**Onewheel (please complete the following information):**
 - Model: Pint
 - Hardware 5302
 - Firmware: 5050

**Additional context**
I installed OWCE for the first time today.

**Comment by beeradmoore** [comment]
> AFAIK, my Pint's firmware is supported?

This is correct, thanks for pointing that out. I have gone and updated the Google Play Store listing.


> I have sometimes received the "OWCE needs help to connect to your board" message trying to connect. 

It is (unfortunately) expected to get this screen when you try use OWCE with this model of board. Essentially we need to official OW app to connect to the board first, exchange its key/token (see [activation FAQ](https://owce.app/faq/unofficial_activating) if you are at all interested in what key/token means, which is why we need to jump through hoops). Then the official app needs to go away as only one app can connect to a board at a time, then there is a small time limit window that OWCE needs to reconnect  in order to keep the board in this "unlocked" state.

Can you try with both apps open. Connect with official app and as soon as you see it say "ACTIVATED" at the bottom and your ride mode changes close this app (not via settings and force close, I believe removing it from that recants list is closing it). Then swap to OWCE (which currently isn't showing the error, just a grey screen) and pull down in about the middle of the screen. The "pull to refresh" option starts scanning for boards. As soon as yours appears tap it to connect and hopefully we get inside this time window. I forget what the time limit is of this but when I play with my Pint next I'll have a look and let you know.

In a future update I'd like to change this error screen so that when you go away and come back it knows to start scanning for boards and connect to the one you tried to connect to before to at least make this a little bit more user friendly.

> I've never seen this message in Android before.

Thats a first for seeing that popup as well. I know there are several situations where the app may crash but none pre-connected. I'll have a hunt around google play and see if I can locate the crashlog to help identify there.

**Comment by verymilan** [comment]
Hi! Before creating a new issue, i'd like to add my findings here, as they might be related:
On iOS-Testflight, I couldn't get the OWCE app to connect by following the instructions – but when i try to keep both open, OWCE shows info about my Pint X for a brief moment until it crashes. I suppose because only one can be open at a time.
I wonder if I should retry and ignore the "scanning..." and try to do this pull to refresh thing later... .

**Comment by beeradmoore** [comment]
@verymilan , Pint X isn't supported. Although I expect it to not work, not for it to crash.
